lib/services/entity_service.dart

```diff
@@ -35,6 +35,28 @@ class EntityService {
     return null;
   }
 
+  /// TODO - document
+  Future<void> updateEntityDataWithEmail(String email, Map<Object, Object?> data) async {
+    final snapshot = await _collection
+        .where(
+          'email',
+          isEqualTo: email,
+        )
+        .get();
+
+    final List<DocumentSnapshot> documents = snapshot.docs;
+
+    if (documents.isNotEmpty) {
+      final DocumentSnapshot documentSnapshot = documents.first;
+      final DocumentReference documentReference = documentSnapshot.reference;
+
+      await documentReference.update(data);
+      ZOLogger.logMessage("Successfully updated entity with email = ${email} and data = ${data}");
+    } else {
+      ZOLogger.logMessage("Unable to find the entity with email = ${email}. Will not update entity.", isError: true);
+    }
+  }
```

Comment:
It is better to update a document directly, as with tokens

So technically this should be enough:

```dart
Future<void> saveAppTermsVersion(String entityId, int version) async {
  return _collection
      .doc(entityId)
      .update({'lastAcceptedAppTermsVersion': version});
}
```

Reply:
Thank you for a good hint, changed